cypress/e2e/gui.cy.js

```diff
@@ -0,0 +1,163 @@
+/// <reference types = "cypress" />
+
+describe('EngageSphere gui test cases', () => {
+  beforeEach(() => {
+      cy.visit('/')
+      cy.get('button')
+      .contains('Accept')
+      .click()
+  });
+
+    it('Keeps the filters when coming back from the customer details view ', () => {
+      cy.get('select[data-testid="size-filter"]')
+      .should('be.visible')
+      .select('Medium')
+      .should('have.value', 'Medium')
+      cy.get('button')
+      .eq(5)
+      .click();
+      cy.get('button')
+      .contains('Back')
+      .click();
+      cy.get('select[data-testid="size-filter"]')
+      .should('have.value', 'Medium')
+    });
+
+    it('It goes back to the customers list when clicking the "Back" button *', () => {
+      cy.get('button')
+      .eq(5)
+      .click();
+      cy.get('button')
+      .contains('Back')
+      .click();
+      cy.get('p').eq(0)
+      .should('have.text', 'Below is our customer list.')
+      .and('be.visible')
+    });
+
+    it('It renders the footer with the right text and links *', () => {
+      cy.get('footer')
+      .should('be.visible')
+      .contains('Copyright 2024 - Talking About Testing')
+      cy.get('a[href="https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem"]')
+      .should('have.text', 'Hotmart')
+      cy.get('a[href="https://udemy.com/user/walmyr" ]')
+      .should('have.text', 'Udemy')
+      cy.get('a[href="https://talkingabouttesting.com" ]')
+      .should('have.text', 'Blog')
+      cy.get('a[href="https://youtube.com/@talkingabouttesting" ]')
+      .should('have.text', 'YouTube')
+    });
+
+    it('It renders the "Hi there" greeting when no name is provided *', () => {
+      cy.get('input[id=name]')
+      .should('have.value', '')
+      cy.get('h2')
+      .should('contain', 'Hi there!')
+    });
+
+    it('It renders the "Hi Joe" greeting when name is provided *', () => {
+      cy.get('input[id=name]')
+      .type('Joe')
+      .should('have.value', 'Joe')
+      cy.get('h2')
+      .should('contain', 'Hi Joe!')
+    });
+
+    it('It renders the header with a heading, theme`s toggle, and a text input field', () => {
+      cy.get('.Header_container__7BmbC')
+      .should('be.visible')
+      .contains('EngageSphere')
+      cy.get('.ThemeToggle_container__d5Yr2')
+      .should('be.visible')
+      cy.get('input[id=name]')
```

Comment:
Tal `cy.get()` pode ser simplificado da seguinte forma:

```suggestion
      cy.get('#name')
```

Reply:
Mais curto e mais simples.

Reply:
Algum motivo por não implementar esta sugestão?